addons/point_of_sale/models/pos_session.py

```diff
@@ -1496,16 +1496,13 @@ def _update_amounts(self, old_amounts, amounts_to_add, date, round=True, force_c
         new_amounts['amount_converted'] += amount_converted
 
         # consider base_amount if present
-        if not amounts_to_add.get('base_amount') == None:
+
+        if amounts_to_add.get('base_amount'):
             base_amount = amounts_to_add.get('base_amount')
-            if self.is_in_company_currency or force_company_currency:
-                base_amount_converted = base_amount
-            else:
-                base_amount_converted = self._amount_converter(base_amount, date, round)
 
             # update base_amount and base_amount_converted
             new_amounts['base_amount'] += base_amount
-            new_amounts['base_amount_converted'] += base_amount_converted
+            new_amounts['base_amount_converted'] += base_amount
```

Comment:
Hey @stpa-odoo ,

Can you tell me the accounting logic behind this change? I'm missing the knowledge for this :sweat_smile:  
I tested your test with and without this part of the fix and in both case it passes successfully so I'd like to understand what it changes.

Reply:
`base_amount` should always be converted. With `rate` now being directly added to the `base_line`, the conversion is made in `prepare_tax_totals`, we need to remove it from `_update_amounts`.
https://github.com/odoo/odoo/blob/1df1ebfa9756b0c213e3365dbab2a66c8353b7b4/addons/point_of_sale/models/pos_session.py#L890-L893 https://github.com/odoo/odoo/blob/811ed888d37024e0056ad4523eae237ab97df024/addons/account/models/account_tax.py#L1342-L1352 https://github.com/odoo/odoo/blob/811ed888d37024e0056ad4523eae237ab97df024/addons/account/models/account_tax.py#L1447 https://github.com/odoo/odoo/blob/811ed888d37024e0056ad4523eae237ab97df024/addons/account/models/account_tax.py#L2175

This change is for `test_05_tax_base_amount` that failed here (the other errors are simply formatting that doesn't really matter): https://runbot.odoo.com/runbot/build/71582022 https://github.com/odoo/odoo/blob/3d922cf4bced230b58e7c8d92ded2c2b8f1ec139/addons/point_of_sale/tests/test_pos_other_currency_config.py#L302-L304 ![image](https://github.com/user-attachments/assets/1634f46c-95e3-436d-bd83-b7f45bad3ff0)

Reply:
I see now, thanks for the explanation ! Looks good to me then :+1: